src/Microsoft.PowerShell.Commands.Utility/commands/utility/MatchString.cs

```diff
@@ -1427,6 +1427,13 @@ protected override void BeginProcessing()
                 this.ThrowTerminatingError(errorRecord);
```

Comment:
@iSazonov, your last commit had 3 failures in `PowerShell-CI-windows`
Enter-PSHostProcess tests.By Process Id.Can enter using NamedPipeConnectionInfo
```powershell
Exception calling "Invoke" with "0" argument(s): "The runspace state is not valid for this operation."
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Core\Enter-PSHostProcess.Tests.ps1: line 136
136:                 $ps.AddScript('$pid').Invoke() | Should -Be $pwshId

```
Select-String.String actions.Should output a string with all matches highlighted when AllMatch is used
```powershell
Cmdlet parameters AllMatches and SimpleMatch are mutually exclusive.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Utility\Select-String.Tests.ps1: line 105
105:                 $result = $testinputone | Select-String -Pattern "l" -AllMatch | Out-String

```
Select-String.String actions.Should throw if -AllMatches parameter is used with -SimpleMatch parameter
```powershell
Expected an exception, with FullyQualifiedErrorId 'CannotSpecifyAllMatchesWithSimpleMatch,Microsoft.PowerShell.Commands.SelectStringCommand' to be thrown, but no exception was thrown.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Utility\Select-String.Tests.ps1: line 111
111:             { "1" | Select-String -Pattern "hello" -AllMatches -SimpleMatch } | Should -Throw -ErrorId "CannotSpecifyAllMatchesWithSimpleMatch,Microsoft.PowerShell.Commands.SelectStringCommand"
```